### PR TITLE
Equipment - Filter loadouts in arsenal

### PR DESCRIPTION
**Pull request by veteran29**
- loadout filtering on arsenal close and template loadout loading
- dependent on gear available to player (`vgm_equipment`)

### CODE REVIEW

File: game/functions/systems/equipment/client/fn_equipment_arsenalInit.sqf [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        Nothing
+
+    Example(s):
+        [parameter] call vgm_c_fnc_equipment_arsenalInit
+ */
+
+[true, "arsenalPreOpen", {
+    // Add our special Medical items to "Misc" tab
+    {BIS_fnc_arsenal_data select 24 pushBackUnique _x} forEach ["vn_helper_item_firstaidkit", "vn_helper_item_medikit"];
+}] call BIS_fnc_addScriptedEventHandler;
```
Comment (Spoffy) [inline game/functions/systems/equipment/client/fn_equipment_arsenalInit.sqf:25]: ❤️

File: game/functions/systems/equipment/client/fn_equipment_filterLoadout.sqf [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    {
+        if (count _x == 7) then {
+            _x call _fnc_checkWeapon;
+            continue;
+        };
+        private _itemClass = _x select 0;
+        if (!(_itemClass in _allowedItems)) then {
+            _items set [_forEachIndex, []];
+        };
+        // TODO is it possible to have container in container?
```
Comment (Spoffy) [inline game/functions/systems/equipment/client/fn_equipment_filterLoadout.sqf]: Yup, turns out you can have vests and uniforms inside backpacks. Here's what it looks like:
```
[[],[],[],[],[],["vn_c_pack_02",[["vn_o_vest_04",false],["vn_b_uniform_macv_03_17",false]]],"","",[],["vn_b_item_map","","vn_b_item_radio_urc10","vn_b_item_compass_sog","vn_b_item_watch",""]]
```
probably worth addressing now, as players will definitely find this loophole :D

File: game/functions/systems/equipment/client/fn_equipment_filterLoadout.sqf
Comment (Spoffy) [inline game/functions/systems/equipment/client/fn_equipment_filterLoadout.sqf:1]: To me, `fn_equipment_filterLoadout` to be a little shennanigany, modifying its input array when the functions are called 'check'.

The backpack one where it literally resizes _this threw me off for a minute, until I realised _this was a reference to the backpack entry in the loadout array :D

As a cheap change, can I propose we rename `fn_checkX` functions to `fn_filterX` or `removeForbiddenFromX`? I think it more strongly implies it's doing some work on the input arrays.

I'd probably prefer something like `_loadout set [_x, _newEntry]` in the filter loops, but I don't think it's worth the time spent making that change.

**Review by Spoffy — CHANGES_REQUESTED** [review]
Overall looks good! 

Couple of proposed tweaks :)

File: game/functions/systems/equipment/client/fn_equipment_filterLoadout.sqf
Comment (veteran29) [reply game/functions/systems/equipment/client/fn_equipment_filterLoadout.sqf:1]: > I'd probably prefer something like _loadout set [_x, _newEntry] in the filter loops, but I don't think it's worth the time spent making that change.

It's bit hard to follow but it's a lot easier to just modify via ref, that's why I went with it.

I agree on the naming change, the function originally was called `checkGear` and I forgot to change the internal functions.

**Review by Spoffy — APPROVED** [review]
Awesome, thanks for making those changes, it definitely felt easier to understand! :)
